src/helpers/completion.ts

```diff
@@ -77,11 +77,7 @@ export async function generateCompletion({
     }
 
     const response = error.response;
-    const message =
-      response &&
-      JSON.parse(
-        await streamToString(response.data as unknown as IncomingMessage)
-      );
+    const message = error.response?.data;
```

Comment:
so this is what the code actually was prior, and most of the type the error is JSON and sent as a string (for instance the most common error - a 429 from openai saying that you have hit your limits or don't have a paid plan) so I think we need to handle both cases, perhaps by checking the type of `response.data` and handling a stream if it is a stream, or handling it as a string if it is a string